lib/Wikia/src/Service/User/Permissions/PermissionsServiceImpl.php

```diff
@@ -283,19 +293,29 @@ private function removeFromLocalGroup( \User $user, $group ) {
 		return true;
 	}
 
-	public function removeFromGroup( \User $performer, \User $userToChange, $group ) {
-		if ( !$this->canGroupBeRemoved( $performer, $userToChange, $group ) ) {
-			return false;
+	public function removeFromGroup( \User $performer, \User $userToChange, $groups )
+	{
+		$groupList = $groups;
+		if ( !is_array( $groups ) ) {
+			$groupList = [ $groups ];
 		}
 
-		try {
-			if ( in_array( $group, $this->permissionsConfiguration->getGlobalGroups() ) ) {
-				$result = $this->removeFromGlobalGroup( $userToChange, $group );
-			} else {
-				$result = $this->removeFromLocalGroup( $userToChange, $group );
+		//First check if we can remove all groups (if we remove in parallel to checking, then the check may not be valid)
+		foreach ( $groupList as $group ) {
+			if ( !$this->canGroupBeRemoved( $performer, $userToChange, $group ) ) {
+				return false;
 			}
 		}
-		finally {
+		$result = true;
+		try {
+			foreach ( $groupList as $group ) {
+				if ( in_array( $group, $this->permissionsConfiguration->getGlobalGroups() ) ) {
+					$result = $this->removeFromGlobalGroup( $userToChange, $group ) && $result;
+				} else {
+					$result = $this->removeFromLocalGroup( $userToChange, $group ) && $result;
+				}
+			}
+		} finally {
```

Comment:
If it's only DB-related exception that can be thrown here, then we're fine. Their constructor reports them to ELK.